compiler/rustc_mir/src/monomorphize/partitioning/default.rs

```diff
@@ -354,7 +355,12 @@ fn compute_codegen_unit_name(
     *cache.entry((cgu_def_id, volatile)).or_insert_with(|| {
         let def_path = tcx.def_path(cgu_def_id);
 
-        let components = def_path.data.iter().map(|part| part.data.as_symbol());
+        let components = def_path.data.iter().map(|part| match part.data.get_name() {
+            DefPathDataName::Named(name) => name,
+            DefPathDataName::Anon { namespace } => {
+                Symbol::intern(&format!("{{{{{}}}}}", namespace))
+            }
```

Comment:
Interestingly, it should be impossible to hit this, because of the logic above, requiring only `DefKind::Mod` components.